syntaxes/GDScript.tmLanguage.json

```diff
@@ -286,7 +286,7 @@
         "in_keyword": {
             "patterns": [
                 {
-                    "match": "\\b(?<=for\\s[\\w]*\\s)(in)\\b",
+                    "match": "\\b(?:for\\s[\\w]*\\s)(in)\\b",
                     "name": "keyword.control.gdscript"
```

Comment:
I didn't test this, but I suspect that it will now consider the whole thing as a keyword.
E.g. `for i in array` then `for i in` is a keyword. It shouldn't mark the loop variable (`i` in this case) as a keyword.

The correct thing to do here is to join this with the `for` keyword so that one rule recognizes both keywords. You don't need to remove `for` from `control_flow` (`for` should still be a keyword even before `in` is typed) as long as this is before so it has higher priority; but change this rule to something along the lines of:
```json
{
  "match": "\\b(for)\\s+([a-zA-Z_]\\w*)\\s+(in)\\b",
  "captures": {
    "1": { "name": "keyword.control.gdscript" },
    "2": { "name": "variable.name.gdscript" },
    "3": { "name": "keyword.control.gdscript" }
  }
},
```
I changed `\w` (which matches `[a-zA-Z0-9_]` including numbers) since I believe identifiers can't start with a number (correct me if I'm wrong). It's probably worth checking this on other places in the code too. I'm assuming identifiers allow ASCII only.
I also allowed more than one space in the syntax.
Btw, using a tool like https://regex101.com/ while you write it really helps with understanding nuances in regexes and checking that it will really do what is expected, and prevent against corner cases.

Reply:
I cannot believe that I missed this. You were correct, `i` was purple.

![image](https://user-images.githubusercontent.com/18042232/190680634-f2fc1080-927f-4915-bb83-3312ce0b2c72.png)

Your rule worked great. I removed the middle capture group and it's name block, because we don't actually use the `variable.name` tag for variables. Python doesn't tag variable names, they're just white, so that's the behavior I want to imitate.

> I changed \w (which matches [a-zA-Z0-9_] including numbers) since I believe identifiers can't start with a number (correct me if I'm wrong). It's probably worth checking this on other places in the code too. I'm assuming identifiers allow ASCII only.

Many other places used the full form `[a-zA-Z_][a-zA-Z_0-9]*`, but I just replaced them with `[a-zA-Z_]\\w*`. I wish we could give a block like this a custom name so we don't have to copy it everywhere. 

> Btw, using a tool like [regex101.com](https://regex101.com/) while you write it really helps with understanding nuances in regexes and checking that it will really do what is expected, and prevent against corner cases.

I started off doing this, but having to add all the extra escapes to move a rule over to vscode was too annoying so I stopped. I've thought about switching to yaml files, since the regex strings won't need escapes, but there was some problem with that too.

Reply:
>> It's probably worth checking this on other places in the code too.

> Many other places used the full form `[a-zA-Z_][a-zA-Z_0-9]*`, but I just replaced them with `[a-zA-Z_]\\w*`.

I was actually referring to places capturing identifiers with just `\w+`, which would allow names starting with numbers, e.g. `1whatever`, `0e2f`, `0xFACADE`. I took a look and saw it in 2 places.
So unless this is on purpose for some reason, I would suggest:

[Line 360](https://github.com/godotengine/godot-vscode-plugin/pull/416/files#diff-2a31f168b0e81523889b916085b33a7c7a235830d79b2dc783b261579f88eb02R360):
```json
"match": "(:)\\s*([a-zA-Z_]\\w*)?",
```
[Line 375](https://github.com/godotengine/godot-vscode-plugin/pull/416/files#diff-2a31f168b0e81523889b916085b33a7c7a235830d79b2dc783b261579f88eb02R375):
```json
"match": "(setget)\\s+([a-zA-Z_]\\w*)?(?:[,]\\s*([a-zA-Z_]\\w*)?)?",
```

Reply:
Another good catch, thanks. I found some other places where identifier names were being matched with `[[:alpha:]_]\\w*`, so I standardized all of them to `[a-zA-Z_]\\w*`.